Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/features/settings/ui/viewholder/InputStringSettingViewHolder.java

```diff
@@ -0,0 +1,85 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+package org.dolphinemu.dolphinemu.features.settings.ui.viewholder;
+
+import android.graphics.drawable.Drawable;
+import android.text.TextUtils;
+import android.view.View;
+import android.widget.TextView;
+
+import androidx.annotation.Nullable;
+
+import org.dolphinemu.dolphinemu.R;
+import org.dolphinemu.dolphinemu.features.settings.model.view.InputStringSetting;
+import org.dolphinemu.dolphinemu.features.settings.model.view.SettingsItem;
+import org.dolphinemu.dolphinemu.features.settings.ui.SettingsAdapter;
+
+public final class InputStringSettingViewHolder extends SettingViewHolder
+{
+  private InputStringSetting mInputString;
+  private SettingsItem mItem;
+
+  private TextView mTextSettingName;
+  private TextView mTextSettingDescription;
+
+  private Drawable mDefaultBackground;
+
+  public InputStringSettingViewHolder(View itemView, SettingsAdapter adapter)
+  {
+    super(itemView, adapter);
+  }
+
+  @Override
+  protected void findViews(View root)
+  {
+    mTextSettingName = root.findViewById(R.id.text_setting_name);
+    mTextSettingDescription = root.findViewById(R.id.text_setting_description);
+
+    mDefaultBackground = root.getBackground();
+  }
+
+  @Override
+  public void bind(SettingsItem item)
+  {
+    mInputString = (InputStringSetting) item;
+    mItem = item;
+
+    String inputString = mInputString.getSelectedValue(getAdapter().getSettings());
+
+    itemView.setBackground(mDefaultBackground);
```

Comment:
What is the purpose of this?

Reply:
for itemView.setBackground(mDefaultBackground); i don't know, i think it's from a copy-paste, it seems to be fine without it

Reply:
You probably copy-pasted it from FilePickerViewHolder. That class has the ability to make the background red if something is wrong with the setting, and if it does that, it needs to be able to set the background back to the default later. But you never change the background, so you never need to explicitly set it to the default.